arch/x86/chunkset_avx.c

```diff
@@ -45,6 +90,42 @@ static inline void storechunk(uint8_t *out, chunk_t *chunk) {
     _mm256_storeu_si256((__m256i *)out, *chunk);
 }
 
+static inline chunk_t GET_CHUNK_MAG(uint8_t *buf, uint32_t *chunk_rem, uint32_t dist) {
+    lut_rem_pair lut_rem = perm_idx_lut[dist - 3];
+    __m256i perm_vec, ret_vec;
+    /* While technically we only need to read 4 or 8 bytes into this vector register
+     * for a lot of cases, GCC is compiling this to a shared load for all branches,
+     * preferring the simpler code.  Given that the buf value isn't in GPRs to begin with,
+     * the 256 bit load is _probably_ just as inexpensive */
+    *chunk_rem = lut_rem.remval;
+
+#ifdef MEMORY_SANITIZER
+    /* See note in chunkset_sse4.c for why this is ok */
+    __msan_unpoison(buf+dist, 32 - dist);
```

Comment:
This is interesting. There is another place where this could be used.
- https://github.com/zlib-ng/zlib-ng/blob/e43e0b21893b55d6b44d3f8b7105fcb655e87f5a/inflate.c#L208

Reply:
Actually nevermind, I thought it was zeroing the whole window.

Reply:
It does though, seem like the exact same situation.

Reply:
Maybe.  Is that being initialized to 0 just to prevent msan from flagging it?  In this case we're unpoisoning the memory because we need to do the full load but we know we're only using the first n bytes of it after the shuffle instruction.

Reply:
Yeah I think it is the same. I have put together a commit https://github.com/nmoinvaz/zlib-ng/commit/80e12c06b3e204b335241455d85159d8330295e9.

Reply:
Ah yeah, definitely basically the same exact use case.  And this way, you don't have write to that memory when not being built with sanitizers :).  Though, it'd be better to only unpoison it just before the read you know is doing this rather than just doing it preemptively.  That way you catch if you unintentionally read that part of hte window when you don't expect it.

Reply:
> Though, it'd be better to only unpoison it just before the read you know is doing this rather than just doing it preemptively

That would be better however it would be a lot more tricky.